### PR TITLE
Simplify FlView

### PR DESCRIPTION
**Pull request by robert-ancell**
This PR simplifies the FlView GL interactions. It removes some code paths that were never being used (possibly for future use with platform views?) and complex internal APIs that could be done in a single method call. This is motivated in investigating https://github.com/ubuntu-flutter-community/software/issues/868.

I plan to take this further, since there doesn't seem to be any reason why multiple FlGLArea widgets are used - we can draw all the layers in one widget.

For reviewers - the individual commits aren't particularly interesting, I just did it that way to be sure I wasn't breaking anything along the way and help to diagnose any regressions in the future.

The structural changes that this PR is simplifying came in f94cf140c52c0cc3769d9281d5579e935886fbac.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by robert-ancell** [comment]
@cbracken @gspencergoog do you know how to get a Linux app to have multiple layers so I can confirm that still renders and will render with the proposed single FlGLArea change?

**Comment by robert-ancell** [comment]
@jpnurmi I don't think this change will specifically fix the issue you were seeing, but can you test in case it does?

**Comment by jpnurmi** [comment]
I could still see a blank window occasionally slip through with this but not with #39473.

**Comment by whiskeyPeak** [comment]
For what it's worth, this PR, along with adding 
```
char envString[] = "GDK_GL=software-draw-surface";
putenv(envString);
```
in the main function fixes the flickering that I have been experiencing in https://github.com/flutter/flutter/issues/120254

File: shell/platform/linux/fl_view.cc [hunk 1]
```diff
@@ -741,14 +705,14 @@ static void fl_view_size_allocate(GtkWidget* widget,
...
+    GtkWidget* w = reinterpret_cast<GtkWidget*>(iterator->data);
+    if (!gtk_widget_get_visible(w)) {
       continue;
     }
 
-    GtkAllocation child_allocation = child->geometry;
+    GtkAllocation child_allocation = {0, 0, 0, 0};
     GtkRequisition child_requisition;
-    gtk_widget_get_preferred_size(child->widget, &child_requisition, NULL);
+    gtk_widget_get_preferred_size(w, &child_requisition, NULL);
```
Comment (cbracken) [inline shell/platform/linux/fl_view.cc:715]: Teensy style nit:
```suggestion
    gtk_widget_get_preferred_size(w, &child_requisition, nullptr);
```

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Pull request merged**